### PR TITLE
Docusaurus Updates

### PR DESCRIPTION
**Pull request by jenningsanderson**
# Description

**House-keeping 🧹:**

1. Bumps docusaurus and plugins to the latest versions.

2. Removes the need for hacky `YAML_FILE_TREE.js` file. Loads _all_ contents of `schema` instead in a more _react_-y way.

**Documentation Changes 👀 :**

3. Returns feature-model, naming, scoping, and transportation pages to the schema reference. 

**Build Changes 🚧 :**

4. Stops publishing the _docusaurus_ folder to `schema.overturemaps.org`. The official documentation page at docs.overturemaps.org is built from the `schema` repo @ `main`, using these paths: 
- `schema/examples`
- `schema/schema`
- `schema/docusaurus`

When a PR is submitted to the `schema` repo, however, the existing behavior is preserved. It will build an example docusaurus page available at `https://dfhx9f55j8eg5.cloudfront.net/pr/` for the author to review and ensure it builds properly.

5. Since the subdomain `schema.overturemaps.org` points to this repository's github page, we can start publishing the machine-readable `schema` folder on the `main` branch. 

# Reference

- https://github.com/OvertureMaps/schema/issues/161
- https://github.com/OvertureMaps/docs/pull/46

# Testing

https://dfhx9f55j8eg5.cloudfront.net/pr/191

# Checklist

*Checklist of tasks commonly-associated with schema pull requests. Please review the relevant checklists and ensure you do all the tasks that are required for the change you made.*

1. [N/A] Add relevant examples.
2. [N/A] Add relevant counterexamples.
4. [N/A] Update any counterexamples that became obsolete. For example, if a counterexample uses property `A` but is not intended to test property `A`'s validity, and you made a schema change that invalidates property `A` in that counterexample, fix the counterexample to align it with your schema change.  
8. [N/A] Update in-schema documentation using plain English written in complete sentences, if an update is required.
9. [N/A] Update Docusaurus documentation, if an update is required.
10. [N/A] Review change with Overture technical writer to ensure any advanced documentation needs will be taken care of, unless the change is trivial and would not affect the documentation.

Note, docs PR: https://github.com/OvertureMaps/docs/pull/46 depends on this PR.

# Documentation Website

[Docs preview for this PR.](https://dfhx9f55j8eg5.cloudfront.net/pr/191)

### CODE REVIEW

File: docusaurus/docs/schema/0-Schema.mdx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import JSONSchemaViewer from "@theme/JSONSchemaViewer";
+import generateResolverOptions from "@site/src/components/shared-libs/generateResolverOptions"
+import StringifyObject from "@site/src/components/shared-libs/stringifyObject" 
+import yamlLoad from "@site/src/components/yamlLoad" 
+import BuildingFeature from "!!raw-loader!@site/docs/_examples/buildings/empire-state-building.yaml";
+import MainDefs from "!!raw-loader!@site/docs/_schema/defs.yaml";
+
+Overture Maps data is structured by three components: the schema, the data model, and GERS. The schema describes the shape of the data and devises the constraints applied to that data. The data model specifies what types of features exist, their geometries, how the features relate to each other, and what kind of attributes they have. Every feature in Overture Maps has a unique ID, and for some features types that unique ID is part of the Global Entity Reference System, a system of encoding map data to a shared universal reference. You can see all three components in this description of one feature, the Empire State Building:
+
+ <CodeBlock language="json">{ JSON.stringify(yamlLoad(BuildingFeature), null, 2) }</CodeBlock>
```
Comment (danabauer) [inline docusaurus/docs/schema/0-Schema.mdx:19]: @jenningsanderson I'm not sure this is the correct way to reference that yaml file

**Review by vcschapp — APPROVED** [review]
Some miscellaneous comments:

1. Should `schema/docusaurus` be re-pathed to `schema/content` if it's going to be a live path on the docs website?
2. Ideally we should try to vertically compress the size of the Empire State Building GeoJSON example on the main page. It takes up so much vertical space it kind of makes the page feel empty and like a data dump instead of documentation.
3. Regarding this sentence: *You can see all three components in this description of one feature, the Empire State Building*, should there be any further elaboration there, or is it obvious what the three components are? My gut is that elaboration would be nice.

File: docusaurus/docs/schema/0-Schema.mdx (hunk 1 above)
Comment (jenningsanderson) [reply docusaurus/docs/schema/0-Schema.mdx]: Yep! This is fixed now and the file is JSON

**Comment by jenningsanderson** [comment]
> 1. Should `schema/docusaurus` be re-pathed to `schema/content` if it's going to be a live path on the docs website?

Good question — it's not a live path, it becomes `docs.overturemaps.org/schema`. But yeah, if renaming it to `content` is less awkward than `docusaurus`. 

> 2. Ideally we should try to vertically compress the size of the Empire State Building GeoJSON example on the main page. It takes up so much vertical space it kind of makes the page feel empty and like a data dump instead of documentation.

Done!

**Review by DavidKarlas — APPROVED** [review]
Json now looks good!

**Review by vcschapp — APPROVED** [review]

**Review by danabauer — APPROVED** [review]

**Review by DavidKarlas — APPROVED** [review]

**Review by TristanDiet-TomTom — APPROVED** [review]

**Comment by jenningsanderson** [comment]
This branch `docusaurus-updates` is being incorporated into the production build of `docs.overturemaps.org` — please do not delete after merging. Once `dev` is merged into staging/main, we can update `docs` to pull from there.

**Pull request merged**